package/MDAnalysis/core/topologyobjects.py

```diff
@@ -124,6 +124,15 @@ def __iter__(self):
     def __len__(self):
         return len(self.atoms)
 
+    def _cmp_key(self):
+        """Unique key for the object to be used to generate the object hash"""
+        # This key must be equal for two object considered as equal by __eq__
+        return '_'.join(map(str, sorted(self.indices)))
+
+    def __hash__(self):
+        """Makes the object hashable"""
+        return hash(self._cmp_key())
```

Comment:
I don't think a hash needs to be a string, so you could just do `hash(self.indices)` or maybe `hash((self.__class__.__name__, self.indices))`.  I think this hash will happen a lot, so I'd rather it was fast (not constructing strings each time)